### DISCUSSION TITLE
Best way to rename a prisma model without data loss?

### DISCUSSION THREAD

**Discussion by zomars** [prisma/prisma#11170] [Q&A]
May be related to #7710 #8381 #8407

Having a model like this:

```
model Credential {
   id     Int    @id @default(autoincrement())
  # ...
}
```

and renaming it to:

```
model NotCredential {
   id     Int    @id @default(autoincrement())
  # ...
}
```

It generates a migration like this

```
-- DropTable
 DROP TABLE "Credential";

 -- CreateTable
 CREATE TABLE "NotCredential" (
     "id" SERIAL NOT NULL,
     CONSTRAINT "NotCredential_pkey" PRIMARY KEY ("id")
 );
```

Is there a way to let prisma know that this is the same model but just got renamed?

**Comment by PeerRich** [comment]
pushing! this issue caused us to lose some staging data (luckily the migration didn't make it to production)

**Reply by floelhoeffel** [reply]
Unfortunately this not trivial to solve. Hopefully we get to it, as it's clearly on our radar. 

Migrate should warn you about the table drop. If those are ignored, the only way to catch these is by reviewing the SQL that migrate generate.

@PeerRich could you elaborate how you generate your migrations? Is there a human present or do you generate them in CI (so the warning about the potentially destructive operation could have been missed)?

**Reply by zomars** [reply]
We have "post build" script that runs migrations in Vercel previews. In this case we totally missed the drop statement.

**Reply by floelhoeffel** [reply]
Interesting - and locally when developing the change you only use `db push` and then `migrate dev` in CI?

**Reply by zomars** [reply]
migrate deploy on CI @floelhoeffel

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
